### PR TITLE
MAGN-9813 API Documentation Priority 2

### PR DESCRIPTION
**Pull request by pbidenko**
### Purpose

Documentation got added to Priority 2 classes/interfaces.

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@ramramps 

### FYIs

@marimano

### CODE REVIEW

**Comment by marimano** [comment]
The changes from this PR are included here - https://github.com/DynamoDS/Dynamo/issues/6421

File: src/DynamoCore/Configuration/Configurations.cs [hunk 1]
```diff
@@ -3,6 +3,9 @@
 
 namespace Dynamo.Configuration
 {
+    /// <summary>
+    ///     This class contains the constants to configure some aspects of Dynamo's work.
```
Comment (ramramps) [inline src/DynamoCore/Configuration/Configurations.cs]: This class contains properties that are used in Dynamo.

File: src/DynamoCore/Core/AuthenticationManager.cs [hunk 2]
```diff
@@ -8,6 +8,10 @@
 
 namespace Dynamo.Core
 {
+    /// <summary>
+    ///     This is a wrapper for <see cref="IAuthProvider"/> functionality.
+    ///     It's used for oxygen authentication.
```
Comment (ramramps) [inline src/DynamoCore/Core/AuthenticationManager.cs]: It's = It is

File: src/DynamoCore/Extensions/ExtensionLoader.cs [hunk 3]
```diff
@@ -12,6 +12,9 @@
 
 namespace Dynamo.Extensions
 {
+    /// <summary>
+    /// Provides functionality for loading Dynamo's extensions
```
Comment (ramramps) [inline src/DynamoCore/Extensions/ExtensionLoader.cs]: Provide some example here.

File: src/DynamoCore/Extensions/IExtensionLoader.cs [hunk 4]
```diff
@@ -10,7 +10,14 @@ namespace Dynamo.Extensions
     /// </summary>
     public interface IExtensionLoader
     {
+        /// <summary>
+        /// Loads assembly by passed extension path and return it as IExtension
```
Comment (ramramps) [inline src/DynamoCore/Extensions/IExtensionLoader.cs]: Loads assembly from the specified path.

File: src/DynamoCore/Extensions/IExtensionLoader.cs [hunk 5]
```diff
@@ -10,7 +10,14 @@ namespace Dynamo.Extensions
...
+        /// </summary>
         IExtension Load(string extensionPath);
+
+        /// <summary>
+        /// Returns an enum of IExtension specified by passed extension path
```
Comment (ramramps) [inline src/DynamoCore/Extensions/IExtensionLoader.cs]: Extension method for loading assembly from a directory. Returns cref = IExtension.

File: src/DynamoCore/Extensions/StartupParams.cs [hunk 6]
```diff
@@ -34,6 +34,9 @@ public class StartupParams
         public Version DynamoVersion { get { return dynamoVersion; } }
         private readonly Version dynamoVersion;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StartupParams"/> class.
```
Comment (ramramps) [inline src/DynamoCore/Extensions/StartupParams.cs]: Specify the usage here.

File: src/DynamoCore/Graph/Annotations/AnnotationModel.cs [hunk 7]
```diff
@@ -9,6 +9,9 @@
 
 namespace Dynamo.Graph.Annotations
 {
+    /// <summary>
+    /// This class represents a model of Dynamo elements' group
```
Comment (ramramps) [inline src/DynamoCore/Graph/Annotations/AnnotationModel.cs]: This class contains methods and properties used for creating groups in Dynamo.

File: src/DynamoCore/Graph/Connectors/ConnectorModel.cs [hunk 8]
```diff
@@ -5,8 +5,14 @@
 
 namespace Dynamo.Graph.Connectors
 {
+    /// <summary>
+    /// Idicates which type of connector's should be displayed on canvas.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Connectors/ConnectorModel.cs]: Returns the Type of connector.

File: src/DynamoCore/Graph/Connectors/ConnectorModel.cs [hunk 9]
```diff
@@ -5,8 +5,14 @@
...
+    /// </summary>
     public enum ConnectorType { BEZIER, POLYLINE };
 
+    /// <summary>
+    /// Represents a connector between nodes. Connector can be a bezier or polyline type.
+    /// </summary>
```
Comment (ramramps) [inline src/DynamoCore/Graph/Connectors/ConnectorModel.cs]: Use a cref = connectortype.

File: src/DynamoCore/Graph/Nodes/CodeBlockUtils.cs [hunk 10]
```diff
@@ -6,6 +6,9 @@
 
 namespace Dynamo.Graph.Nodes
 {
+    /// <summary>
+    /// Contains helper methods to adjust content of CodeBlock node.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/CodeBlockUtils.cs]: Contains helper methods to adjust the contents of codeblock node.

File: src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs [hunk 11]
```diff
@@ -245,6 +252,9 @@ public void ResyncWithDefinition(CustomNodeDefinition def)
         }
     }
 
+    /// <summary>
+    ///     Represents input of Custom Node instance.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs]: Incorrect. You have to explain the usage of Symbol class.

File: src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs [hunk 12]
```diff
@@ -439,6 +449,9 @@ protected override bool UpdateValueCore(UpdateValueParams updateValueParams)
         }
     }
 
+    /// <summary>
+    /// Represents output of CustomNode
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs]: Incorrect. You have to explain the usage of the class

File: src/DynamoCore/Graph/Nodes/NodeModel.cs [hunk 13]
```diff
@@ -2112,6 +2112,9 @@ private void OnRenderPackagesUpdated(IEnumerable<IRenderPackage> packages)
         }
     }
 
+    /// <summary>
+    /// Enumeration of possible Node's states
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/NodeModel.cs]: Represents nodes states.

File: src/DynamoCore/Graph/Workspaces/CustomNodeWorkspaceModel.cs [hunk 14]
```diff
@@ -15,6 +15,9 @@
 
 namespace Dynamo.Graph.Workspaces
 {
+    /// <summary>
+    /// Extension of standard WorkspaceModel. Represents workspace of CustomNode inner functionality.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Workspaces/CustomNodeWorkspaceModel.cs]: Represents workspace of a CustomNode

File: src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs [hunk 15]
```diff
@@ -18,6 +18,9 @@
 
 namespace Dynamo.Graph.Workspaces
 {
+    /// <summary>
+    ///     This class represents executable workspace.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs]: This class contains methods and properties that defines a (use cref) workspacemodel.

File: src/DynamoCore/Graph/Workspaces/IWorkspaceModel.cs [hunk 16]
```diff
@@ -5,6 +5,9 @@
 
 namespace Dynamo.Graph.Workspaces
 {
+    /// <summary>
+    /// An interface of Workspace Model
```
Comment (ramramps) [inline src/DynamoCore/Graph/Workspaces/IWorkspaceModel.cs]: Exposes workspace model.

File: src/DynamoCore/Library/ILibraryLoader.cs [hunk 17]
```diff
@@ -2,8 +2,14 @@
 
 namespace Dynamo.Library
 {
+    /// <summary>
+    /// An interface which provides functionality for loading node's library
```
Comment (ramramps) [inline src/DynamoCore/Library/ILibraryLoader.cs]: Exposes (use cref) LoadNodeLibrary method. Specify the usage of LoadNodeLibrary

File: src/DynamoCore/Logging/DynamoLogger.cs [hunk 18]
```diff
@@ -61,6 +61,9 @@ public LogEventArgs(Exception e, LogLevel level)
         }
     }
 
+    /// <summary>
+    /// Contains logging functionality.
```
Comment (ramramps) [inline src/DynamoCore/Logging/DynamoLogger.cs]: This class contains methods and properties used for logging in Dynamo,

File: src/DynamoCore/Migration/Migration.cs [hunk 19]
```diff
@@ -41,6 +41,9 @@ public Migration(Version v, Action upgrade)
         }
     }
 
+    /// <summary>
+    /// A class which provides functionality for managing migration of Dynamo workspaces
```
Comment (ramramps) [inline src/DynamoCore/Migration/Migration.cs]: This class contains methods and properties used for migration of Dynamo workspaces.

File: src/DynamoCore/Search/BrowserInternalElement.cs [hunk 20]
```diff
@@ -2,6 +2,9 @@
 
 namespace Dynamo.Search
 {
+    /// <summary>
+    ///     This class represents "descendant" of browser elemet
```
Comment (ramramps) [inline src/DynamoCore/Search/BrowserInternalElement.cs]: descendant ?? Browser element?

File: src/DynamoCore/Search/BrowserItem.cs [hunk 21]
```diff
@@ -9,6 +9,9 @@
 
 namespace Dynamo.Search
 {
+    /// <summary>
+    ///     Base class for all browser items in search area.
```
Comment (ramramps) [inline src/DynamoCore/Search/BrowserItem.cs]: provide example for browser items?

File: src/DynamoCore/Search/SearchElements/NodeSearchElement.cs [hunk 22]
```diff
@@ -278,6 +278,9 @@ protected virtual IEnumerable<string> GenerateOutputParameters()
         }
     }
 
+    /// <summary>
+    /// This class is being used when drag-drop search element on canvas to create a node.
```
Comment (ramramps) [inline src/DynamoCore/Search/SearchElements/NodeSearchElement.cs]: This class returns (use cref) NodeSearchElement which is used for creating a node.

File: src/DynamoCore/Search/SearchElements/NodeSearchElement.cs [hunk 23]
```diff
@@ -278,6 +278,9 @@ protected virtual IEnumerable<string> GenerateOutputParameters()
...
+    /// </summary>
     public class DragDropNodeSearchElementInfo
     {
         public NodeSearchElement SearchElement { get; private set; }
```
Comment (ramramps) [inline src/DynamoCore/Search/SearchElements/NodeSearchElement.cs]: Comments?

File: src/DynamoCore/Updates/BinaryVersion.cs [hunk 24]
```diff
@@ -1,5 +1,8 @@
 ﻿namespace Dynamo.Updates
 {
+    /// <summary>
+    /// This class represents binary current version of the product.
```
Comment (ramramps) [inline src/DynamoCore/Updates/BinaryVersion.cs]: remove binary

File: src/DynamoCore/Updates/UpdateManager.cs [hunk 25]
```diff
@@ -15,6 +15,9 @@
 namespace Dynamo.Updates
 {
     public delegate void UpdateDownloadedEventHandler(object sender, UpdateDownloadedEventArgs e);
+    /// <summary>
+    /// A delegate is used to handle shutdown request
```
Comment (ramramps) [inline src/DynamoCore/Updates/UpdateManager.cs]: Remove is.

File: src/DynamoCore/Updates/UpdateManager.cs [hunk 26]
```diff
@@ -126,6 +129,10 @@ public interface IAsynchronousRequest
         Action<IAsynchronousRequest> OnRequestCompleted { get; set; }
     }
 
+    /// <summary>
+    /// A class to describe available
```
Comment (ramramps) [inline src/DynamoCore/Updates/UpdateManager.cs]: This class returns (use cref) Binaryversion of Dynamo

**Pull request closed without merging**